File: articles/algorithmic-thinking-vs-computational-research/java/src/main/java/org/contentcatalyst/algorithms/ReasoningProfiles.java
package org.contentcatalyst.algorithms;

public class ReasoningProfiles {
    public static void main(String[] args) {
        String[] names = {
            "Recipe-like procedure",
            "Classroom algorithm exercise",
            "Search and ranking system",
            "Public decision-support workflow",
            "Scientific modeling workflow"
        };
        double[] step = {86, 90, 72, 68, 74};
        double[] decomp = {72, 82, 70, 66, 78};
        double[] control = {70, 84, 76, 64, 76};
        double[] test = {62, 78, 66, 72, 82};
        double[] representation = {42, 62, 78, 80, 86};
        double[] governance = {20, 32, 70, 86, 74};

        System.out.println("name,algorithmic_score,computational_score,warning");
        for (int i = 0; i < names.length; i++) {
            double algorithmic = 0.28 * step[i] + 0.24 * decomp[i] + 0.24 * control[i] + 0.24 * test[i];
            double computational = 0.16 * step[i] + 0.14 * decomp[i] + 0.14 * control[i] + 0.14 * test[i] + 0.22 * representation[i] + 0.20 * governance[i];
            System.out.printf("%s,%.3f,%.3f,Synthetic educational diagnostic only.%n", names[i], algorithmic, computational);
        }
    }
}

File: articles/algorithmic-thinking-vs-computational-research/prolog/reasoning_profiles.pl
profile('Recipe-like procedure', 86, 72, 70, 62, 42, 20).
profile('Classroom algorithm exercise', 90, 82, 84, 78, 62, 32).
profile('Search and ranking system', 72, 70, 76, 66, 78, 70).
profile('Public decision-support workflow', 68, 66, 64, 72, 80, 86).
profile('Scientific modeling workflow', 74, 78, 76, 82, 86, 74).

algorithmic_score(Name, Score) :-
    profile(Name, Step, Decomp, Control, Test, _, _),
    Score is 0.28*Step + 0.24*Decomp + 0.24*Control + 0.24*Test.

computational_score(Name, Score) :-
    profile(Name, Step, Decomp, Control, Test, Representation, Governance),
    Score is 0.16*Step + 0.14*Decomp + 0.14*Control + 0.14*Test + 0.22*Representation + 0.20*Governance.
